background.js

```
@@ -0,0 +1,443 @@
+    return false;
+  }
+
+  // Check patterns first
+  for (const pattern of settings.patterns) {
+    if (matchesPattern(tab.url, pattern.pattern)) {
+      switch (pattern.action) {
+        case 'keep':
+          return false;
+        case 'duplicate':
+          // Find the newest duplicate tab
+          const exactDuplicates = allTabs.filter(t => t.id !== tab.id && t.url === tab.url);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)